palette/src/arithmetic.rs

```diff
@@ -0,0 +1,117 @@
+//! Macros to implement arithmetic traits on Color spaces.
+
+/// Implement `Add` and `AddAssign` traits for a color space.
+///
+/// Both scalars and color arithmetic are implemented.
+#[macro_export]
```

Comment:
I always forget the details, but I'm pretty sure `#[macro_export]` is unnecessary for internal macros, _but_ the order the modules are declared in the `lib.rs` file matters. Macros and macro modules have to come first. I would suggest putting this module inside the `macros` module that's already in here.

Reply:
Ah, I bet the ordering was tripping me up on macros. I'll take your suggestion and just move them to `macros.rs`, though. I'll also do a bi-cone style implementation for `Hsluv`.